cogs/music.py

```diff
@@ -0,0 +1,584 @@
+            return
+        playing = ctx.voice_client.is_playing()
+        paused = ctx.voice_client.is_paused()
+        if playing or paused:
+            status_embed.add_field(
+                name=f"currently {'playing' if playing else 'paused'} track",
+                value=mk_title_string(self.playing[0]),
+                inline=False,
+            ).add_field(
+                name="volume", value=f"{self.volume_level}%",
+                inline=True).add_field(
+                    name="duration",
+                    value=mk_duration_string(self.playing[0]),
+                    inline=True).add_field(
+                        name="looping",
+                        value="no" if self.looping is None else "yes",
+                        inline=True,
+                    ).set_footer(text=f"submitted by: {self.playing[1]}")
+        else:
+            status_embed.add_field(
+                name="status",
+                value=
+                "not playing but connected [ERROR STATE SHOULD NEVER OCCUR]",
+            )
+        await ctx.send(embed=status_embed)
+
+    @commands.command(aliases=["rs"])
+    async def remove_song(self, ctx, song_index: int):
+        """Remove a song from the song queue by index"""
+
+        if song_index < 0 or len(self.song_queue) <= song_index:
+            await ctx.send(
+                f"supplied index {song_index} is not valid for current queue.")
+            return
+        removed = discord.Embed(
+            colour=random.randint(0, 0xFFFFFF),
+            title=f"removed track at index {song_index}",
+        ).add_field(
+            name="track name",
+            value=mk_title_string(self.song_queue[song_index][0]),
+            inline=False).add_field(
+                name="duration",
+                value=mk_duration_string(self.song_queue[song_index][0]),
+                inline=False).set_footer(text=f"removed by: {ctx.author}")
+        del self.song_queue[song_index]
+        await ctx.send(embed=removed)
+
+    @commands.command(aliases=["iqs"])
+    async def insert_song(self, ctx, song_index: int, *, url: str):
+        """Insert a song into the song queue at a given index"""
+
+        await ctx.trigger_typing()
+        if song_index < 0 or len(self.song_queue) <= song_index:
+            await ctx.send(
+                f"supplied index {song_index} is not valid for current queue.")
+            return
+        try:
+            data = await self.get_info(url)
+        except youtube_dl.utils.DownloadError:
+            await ctx.send("could not find track")
+            return
+        entries = data.get("entries", [data])
+        if not entries:
+            await ctx.send("could not find track")
+            return
+        author_str = str(ctx.author)
+        for track in reversed(entries):
+            self.song_queue.insert(song_index, (track, author_str))
+        if len(entries) > 1:
+            inserted = discord.Embed(
```

Comment:
duplicate code between here and the else branch (with the inserted thing, which could be `{'playlist' if len(entries) > 1 else 'track'}`)

you could define this outside the branch, then add the fields as needed conditionally